### ISSUE TITLE
Did the author make any comparisons on mathematical tasks such as GSM8K?

### ISSUE THREAD

**Issue by Facico** [Chongjie-Si/Subspace-Tuning#9]
Models like PISSA and MiLoRA report very high scores on GSM8K. I would like to know if the author has made comparisons on mathematical benchmarks.

At the same time, I found during reproduction that the high scores reported by PISSA benefit from its fast convergence speed, allowing it to perform very well with a small learning rate. However, when using a normal learning rate, such as 3e-4, PISSA loses its advantage completely. Did the author make similar observations?

**Comment by Chongjie-Si** [comment]
Hi, thanks for your attention on our project. We have not made any comparisons on mathematical benchmarks. However, I personally have tried to use PISSA (or MiLoRA) to fine-tune LLaMA family on commonsense reasoning tasks. Unfortunately, I couldn't reproduce comparable performance. In fact, the performances of these methods are somewhat significantly inferior to other methods like vanilla LoRA.

I have also communicated with other researchers, and they found that in real applications, PISSA and MiLoRA need a larger rank such as 64 to guarantee their performances. Under a small rank configuration, they may easily lose their advantages.

**Comment by Facico** [comment]
Thanks to the author for the reminder. I will try to compare the results under a larger rank. 

However, I noticed that the settings for CommonsenseQA vary among different works. Dora and your approach train based on `commonsense_170k.json` and then test each task separately, while many works train using the respective training sets within CommonsenseQA. Using `commonsense_170k.json` as the training set does not seem to be the conventional approach, does it?

**Comment by Chongjie-Si** [comment]
I'm not quite sure about the difference between Commonsense QA and Commonsense Reasoning task. I believe the commonsense reasoning task encompasses various forms, and the community has developed multiple datasets, including CommonsenseQA, SocialIQA, and the Winograd Schema Challenge. The dataset we are using originates from this particular [work](https://arxiv.org/pdf/2304.01933).

**Comment by Chongjie-Si** [comment]
By the way, in the PEFT papers I have reviewed, most researchers have conducted commonsense reasoning tasks on this dataset.

**Comment by Facico** [comment]
I checked again; it is indeed the case that the setting using the merged 170k for training is more common. Thanks!
